packages/platform-sdk-ada/src/dto/transaction.ts

```diff
@@ -0,0 +1,205 @@
+import { Contracts, DTO, Exceptions } from "@arkecosystem/platform-sdk";
+import { DateTime } from "@arkecosystem/platform-sdk-intl";
+import { BigNumber } from "@arkecosystem/platform-sdk-support";
+
+export class TransactionData extends DTO.AbstractTransactionData implements Contracts.TransactionData {
+	public id(): string {
+		return this.data.id;
+	}
+
+	public blockId(): string | undefined {
+		return undefined;
+	}
+
+	public timestamp(): DateTime | undefined {
+		return DateTime.make(this.data.inserted_at.time);
+	}
+
+	public confirmations(): BigNumber {
+		return BigNumber.make(this.data.confirmations);
+	}
+
+	public sender(): string {
+		throw new Exceptions.NotImplemented(this.constructor.name, "sender");
```

Comment:
This would cause the consumer of the package to always crash if they want to list transactions.

Reply:
@faustbrian I need a bit of help here... Given this transaction, what would you use as sender / recipient? For e.g. `inputs.address`? And for output, we have 2, one for the money sent someone else, and another for the change staying back with us. What would be the way to identify out of those 2 which is which?
```
  {
	"inserted_at": {
	  "height": {
		"quantity": 2292394,
		"unit": "block"
	  },
	  "time": "2021-02-05T15:04:16Z",
	  "epoch_number": 112,
	  "absolute_slot_number": 18168240,
	  "slot_number": 153840
	},
	"status": "in_ledger",
	"withdrawals": [],
	"amount": {
	  "quantity": 25168801,
	  "unit": "lovelace"
	},
	"inputs": [
	  {
		"amount": {
		  "quantity": 30000000,
		  "unit": "lovelace"
		},
		"address": "addr_test1qrhvwtn8sa3duzkm93v5kjjxlv5lvg67j530wyeumngu23lk8ttq8f3gag0h89aepvx3xf69g0l9pf80tqv7cve0l33s4s8xvh",
		"id": "6bf76f4380da8a389ae0a7ecccf1922b74ae11d773ba8b1b761d84a1b4474a4f",
		"assets": [],
		"index": 0
	  }
	],
	"direction": "outgoing",
	"fee": {
	  "quantity": 168801,
	  "unit": "lovelace"
	},
	"outputs": [
	  {
		"amount": {
		  "quantity": 25000000,
		  "unit": "lovelace"
		},
		"address": "addr_test1qzct2hsralem3fqn8fupu90v3jkelpg4rfp4zqx06zgevpachk6az8jcydma5a6vgsuw5c37v0c8j6rlclpqajn2vxsq3rz4th",
		"assets": []
	  },
	  {
		"amount": {
		  "quantity": 4831199,
		  "unit": "lovelace"
		},
		"address": "addr_test1qzfjfm724nv9qz6nfyagmj0j2uppr35gzv5qee8s7489wxlk8ttq8f3gag0h89aepvx3xf69g0l9pf80tqv7cve0l33scc4thv",
		"assets": []
	  }
	],
	"metadata": null,
	"depth": {
	  "quantity": 9216,
	  "unit": "block"
	},
	"id": "35b40547f04963d3b41478fc27038948d74718802c486d9125f1884d8c83a31d",
	"deposit": {
	  "quantity": 0,
	  "unit": "lovelace"
	},
	"mint": []
  },
```

Reply:
I would use the address of the wallet that we are looking up transactions for as the sender and I would ignore the change address and use the actual wallet that received the money.